CupertinoSlider initialisation does not allow values less then zero/greater then 1

### Duplicate Check

- [x] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the bug

TL;DR: Initialisations such as ft.CupertinoSlider(min=2, max=3) or ft.CupertinoSlider(min=0, max=3, value=2) throws errors.

When creating a new CupertinoSlider, it's not possible to set values to be less then zero or greater then one.

For example, you cannot create slider like this: ft.CupertinoSlider(min=0, max=255, value=127). you will get an AssertionError

This behavior seems to be due to the way object initialized. It does it in the order: value, min, max. Than means that the value checker in property fails to set a value, if it's out of the defoult [0.0 ... 1.0] range. For the same reason there does not seem to be a way to init a slider with min value greater then 1.0, even if max key is set and greater then min (it will throw an Assertion for setting min to be greater then current max, wich is 1.0 by defoult)

This is where this logic implemented [https://github.com/flet-dev/flet/blob/main/sdk/python/packages/flet/src/flet/core/cupertino_slider.py#L109]

I mark this as a bug since there are no indecation of this way of initialisation being unsupported, moreover, in the documentation [https://flet.dev/docs/controls/cupertinoslider/] the order of the arguments suggest that it's an expected way to create a slider with nondefault range and set value. If it is an expected behavior i think it should be stated in the docs.

### Code sample

<details open><summary>Code</summary>

```python
import flet as ft

def main(page: ft.Page):
    slider = ft.CupertinoSlider(
        min=0,
        max=255,
        value=127,
    )
    page.add(slider)
    page.update()
ft.app(main)
```

this snippet fails with "AssertionError: value must be less than or equal to max"
</details>


### To reproduce

1. Setup any python env with flet installed
2. Create any CupertinoSlider slider with init values of valid range (min <= value <= max) where eather value key is not in the [0.0 ... 1.0] range or min key is greater then 1


### Expected behavior

Slider created without errors (since the range is valid). 
Instead ether "AssertionError: min must be less than or equal to max" or "AssertionError: value must be less than or equal to max"

### Operating System

Windows

### Operating system details

Windows 11

### Flet version

0.26.0

### Regression

I'm not sure / I don't know

### Suggestions

I would eather expect to have eather:
1) value, max and min to be initialised at the same time (and checked for correctness after all three are set) 
2) Their order of the init to correspond to property order in https://flet.dev/docs/controls/cupertinoslider/
3) This behavior stated in the docs



### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>


### Additional details

My understanding is that it is not really common to do those checks in the controls initialisations. But this case does look very unintuitive and easy to fix. 

I also don't think doing 
```python
slider = ft.CupertinoSlider()
slider.max = 5
sliider.min = 2
slider.value = 4
```
Is a reasonable way to go around this initialisation problem

--- a/packages/flet/lib/src/controls/cupertino_slider.dart
+++ b/packages/flet/lib/src/controls/cupertino_slider.dart
@@ -55,9 +55,7 @@ class _CupertinoSliderControlState extends State<CupertinoSliderControl> {
     double max = widget.control.attrDouble("max", 1)!;
     int? divisions = widget.control.attrInt("divisions");
 
-    debugPrint("CupertinoSliderControl build: ${widget.control.id}");
-
-    double value = widget.control.attrDouble("value", 0)!;
+    double value = widget.control.attrDouble("value", min)!;
     if (_value != value) {
       // verify limits
       if (value < min) {

--- a/sdk/python/packages/flet/src/flet/core/cupertino_slider.py
+++ b/sdk/python/packages/flet/src/flet/core/cupertino_slider.py
@@ -137,7 +137,7 @@ class CupertinoSlider(ConstrainedControl):
     # value
     @property
     def value(self) -> float:
-        return self._get_attr("value", data_type="float", def_value=0.0)
+        return self._get_attr("value", data_type="float", def_value=self.min or 0.0)
 
     @value.setter
     def value(self, value: OptionalNumber):

--- a/sdk/python/packages/flet/src/flet/core/cupertino_slider.py
+++ b/sdk/python/packages/flet/src/flet/core/cupertino_slider.py
@@ -122,6 +122,18 @@ class CupertinoSlider(ConstrainedControl):
     def _get_control_name(self):
         return "cupertinoslider"
 
+    def before_update(self):
+        super().before_update()
+        assert (
+            self.min is None or self.max is None or self.min <= self.max
+        ), "min must be less than or equal to max"
+        assert (
+            self.min is None or self.value is None or (self.value >= self.min)
+        ), "value must be greater than or equal to min"
+        assert (
+            self.max is None or self.value is None or (self.value <= self.max)
+        ), "value must be less than or equal to max"
+
     # value
     @property
     def value(self) -> float:
@@ -129,11 +141,6 @@ class CupertinoSlider(ConstrainedControl):
 
     @value.setter
     def value(self, value: OptionalNumber):
-        if value is not None:
-            if self.min is not None:
-                assert value >= self.min, "value must be greater than or equal to min"
-            if self.max is not None:
-                assert value <= self.max, "value must be less than or equal to max"
         self._set_attr("value", value)
 
     # min
@@ -143,9 +150,6 @@ class CupertinoSlider(ConstrainedControl):
 
     @min.setter
     def min(self, value: OptionalNumber):
-        if value is not None:
-            if self.max is not None:
-                assert value <= self.max, "min must be less than or equal to max"
         self._set_attr("min", value)
 
     # max
@@ -155,9 +159,6 @@ class CupertinoSlider(ConstrainedControl):
 
     @max.setter
     def max(self, value: OptionalNumber):
-        if value is not None:
-            if self.min is not None:
-                assert value >= self.min, "max must be greater than or equal to min"
         self._set_attr("max", value)
 
     # divisions

